lib/rubycritic/core/analysed_module.rb

```diff
@@ -27,6 +28,17 @@ def rating
       @rating ||= Rating.from_cost(cost)
     end
 
+    def lines
+      return @lines if @lines
+      output = `wc -l < "#{path}"`.to_i
+      result = $?.success?
+      if result
+        @lines ||= output
+      else
+        @lines ||= File.open(path).lines.count
+      end
+    end
```

Comment:
Most certainly. I assumed that since wc was an optimized system function it would be faster, but I know spawning new system operations takes some overhead. If it turns out that wc is slower I agree that we should move to Ruby only.



> On Mar 14, 2015, at 10:59 AM, Guilherme Simões <notifications@github.com> wrote:
> 
> In lib/rubycritic/core/analysed_module.rb:
> 
> > @@ -27,6 +28,17 @@ def rating
> >        @rating ||= Rating.from_cost(cost)
> >      end
> >  
> > +    def lines
> > +      return @lines if @lines
> > +      output = `wc -l < "#{path}"`.to_i
> > +      result = $?.success?
> > +      if result
> > +        @lines ||= output
> > +      else
> > +        @lines ||= File.open(path).lines.count
> > +      end
> > +    end
> lines was deprecated in favor of each|each_line.
> 
> I did a little benchmark on Counting number of lines of a file in Ruby and File.read(path).each_line.count won.
> 
> However, wc was so slow that I'm thinking there's something wrong with it. Could you please run the benchmarks as well? If wc is not the fastest, I think we could simplify this to just using Ruby.
> 
> —
> Reply to this email directly or view it on GitHub.
>